setuptools/dist.py

```diff
@@ -607,7 +618,7 @@ def parse_config_files(self, filenames=None, ignore_option_errors=False):
         self._finalize_requires()
         self._finalize_license_files()
 
-    def fetch_build_eggs(self, requires):
+    def fetch_build_eggs(self, requires: _StrOrIter):
```

Comment:
By the time we reach this function the argument passed to `requires` is always `list[str]`.

The trouble happens at the `__init__` function, although the user can pass `_StrOrIter` for any of the requirement-related arguments, by the time the `__init__` function finishes all the requirements are processed into a `list[str]`.

Reply:
I typed the function by what it supports. Which is the same as `setuptools._reqs.parse`. A `list[str]`is a valid type to pass to a `_StrOrIter`, so it doesn't create any typing issue here.